dspy/adapters/chat_adapter.py

```diff
@@ -228,26 +267,96 @@ def type_info(v):
     return {"role": role, "content": joined_messages}
 
 
-def enumerate_fields(fields: dict) -> str:
+def _format_constraint(name: str, value: Union[str, float]) -> str:
```

Comment:
We can move this function and other related ones into `dspy/signatures/field` and import in the adapter to keep adapter code simple.